### PR TITLE
Use auto with template factory functions for smart pointers

### PR DESCRIPTION
**Pull request by Alexander-Wilms**

### CODE REVIEW

File: client/CPlayerInterface.h [hunk 1]
```diff
@@ -192,7 +192,7 @@ class CPlayerInterface : public CGameInterface, public IUpdateable
 	void showInfoDialog(const std::string &text, std::shared_ptr<CComponent> component);
 	void showInfoDialog(const std::string &text, const std::vector<std::shared_ptr<CComponent>> & components = std::vector<std::shared_ptr<CComponent>>(), int soundID = 0);
 	void showInfoDialogAndWait(std::vector<Component> & components, const MetaString & text);
-	void showYesNoDialog(const std::string &text, CFunctionList<void()> onYes, CFunctionList<void()> onNo, const std::vector<std::shared_ptr<CComponent>> & components = std::vector<std::shared_ptr<CComponent>>());
+	auto showYesNoDialog(const std::string &text, CFunctionList<void()> onYes, CFunctionList<void()> onNo, const std::vector<std::shared_ptr<CComponent>> & components = std::vector<std::shared_ptr<CComponent>>());
```
Comment (nullkiller) [inline client/CPlayerInterface.h:195]: Not sure it will compile

File: client/mainmenu/CHighScoreScreen.cpp [hunk 2]
```diff
@@ -118,7 +118,7 @@ void CHighScoreScreen::addHighScores()
 {
 	OBJ_CONSTRUCTION_CAPTURING_ALL_NO_DISPOSE;
 
-	background = std::make_shared<CPicture>(ImagePath::builtin(highscorepage == HighScorePage::SCENARIO ? "HISCORE" : "HISCORE2"));
+	auto = std::make_shared<CPicture>(ImagePath::builtin(highscorepage == HighScorePage::SCENARIO ? "HISCORE" : "HISCORE2"));
```
Comment (nullkiller) [inline client/mainmenu/CHighScoreScreen.cpp:121]: same

File: client/mainmenu/CHighScoreScreen.h [hunk 3]
```diff
@@ -35,7 +35,7 @@ class HighScoreParameter
 class HighScoreCalculation
 {
 public:
-	std::vector<HighScoreParameter> parameters = std::vector<HighScoreParameter>();
+	auto parameters = std::vector<HighScoreParameter>();
```
Comment (nullkiller) [inline client/mainmenu/CHighScoreScreen.h:38]: Works in class declaration?

File: client/mainmenu/CHighScoreScreen.cpp (hunk 2 above)
Comment (Alexander-Wilms) [reply client/mainmenu/CHighScoreScreen.cpp:121]: Yeah, there's still some errors caused by the automated replacement

File: lib/LoadProgress.cpp [hunk 4]
```diff
@@ -54,7 +54,7 @@ void Progress::reset(int s)
 
 void Progress::finish()
 {
-	_progress = _target = std::numeric_limits<Type>::max();
+	auto = _target = std::numeric_limits<Type>::max();
```
Comment (nullkiller) [inline lib/LoadProgress.cpp:57]: Same

**Pull request merged**